agent/pagebroker/posix_copy_engine.cpp

```
@@ -1,14 +1,117 @@
 
 namespace snapshot::pagebroker {
+namespace {
+Path
+StoragePath(const StorageBackend& storage, const Path& storage_root, const char* label)
+{
+  if (!storage.has_filesystem() || storage.filesystem().directory().empty())
+    throw std::invalid_argument(std::string("filesystem ") + label + " is required");
+  const Path path(storage.filesystem().directory());
+  const Path relative = path.lexically_relative(storage_root);
+  if (!path.is_absolute() || path.lexically_normal() != path || relative.empty() ||
+      (relative != "." && relative.string().starts_with("../")) || relative == "..")
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)